src/malloc/mallocng/glue.h

```diff
@@ -39,55 +39,94 @@
 
 #define DISABLE_ALIGNED_ALLOC (__malloc_replaced && !__aligned_alloc_replaced)
 
+#ifdef __CHEERP__
+__attribute__((cheerp_asmjs))
+#endif
 static inline uint64_t get_random_secret()
 {
-	uint64_t secret = (uintptr_t)&secret * 1103515245;
-	for (size_t i=0; libc.auxv[i]; i+=2)
-		if (libc.auxv[i]==AT_RANDOM)
-			memcpy(&secret, (char *)libc.auxv[i+1]+8, sizeof secret);
-	return secret;
+	return 6148914691236517205; // represented in binary as alternating 0 and 1.
 }
 
 #ifndef PAGESIZE
 #define PAGESIZE PAGE_SIZE
 #endif
 
+#if defined(__CHEERP__) && !defined(__ASMJS__)
+#define MT (0)
+#else
 #define MT (libc.need_locks)
+#endif
 
 #define RDLOCK_IS_EXCLUSIVE 1
 
 __attribute__((__visibility__("hidden")))
+#ifdef __CHEERP__
+__attribute__((cheerp_asmjs))
+#endif
 extern int __malloc_lock[1];
 
 #define LOCK_OBJ_DEF \
 int __malloc_lock[1]; \
 void __malloc_atfork(int who) { malloc_atfork(who); }
 
+#ifdef __CHEERP__
+__attribute__((cheerp_asmjs))
+#endif
 static inline void rdlock()
 {
+#if 1
+#else
```

Comment:
what about these #if 1?

Reply:
I noticed this construction in the oldmalloc. I thought since we're not using threading I could apply it here aswell. They did not compile at the time, due to MT (which I changed later). They do currently compile when I remove these if blocks.
Should I remove them?

Reply:
I would remove these if it compiles. They will still disappear in genericjs (since the MT=0). For wasm, it will make the code a bit worse, hopefully not much. You could keep them in a separate commit in a branch outside of the PR. Maybe Carlo could check if it makes a noticeable difference.
Eventually with threads we will need this code though....

Reply:
Removed the if blocks